include/onnxruntime/core/common/logging/logging.h

```diff
@@ -58,15 +58,42 @@ namespace logging {
 
 using Timestamp = std::chrono::time_point<std::chrono::system_clock>;
 
-// TODO: When other compilers support std::chrono::operator<<, update this.
-// TODO: Check support for other compilers' version before enable C++20 for other compilers.
-// Xcode added support for C++20's std::chrono::operator<< in SDK version 14.4.
-#if __cplusplus >= 202002L && __MAC_OS_X_VERSION_MAX_ALLOWED >= 140400L
+// C++20 has operator<< in std::chrono for Timestamp type but mac builds need additional checks
+// to ensure usage is valid.
+// TODO: As we enable C++20 on other platforms we may need similar checks.
+#define _USE_CXX20_STD_CHRONO __cplusplus >= 202002L
```

Comment:
nit: the macro name looks like a compiler internal macro.

Reply:
It's a temporary define which is undef'd as soon as used. Is there a better format to use?

Reply:
I think `ONNXRUNTIME_SOMETHING` would be better, in case a compiler or std library suddenly use this name.  Anyway, it is not a big deal. Now it is fine.

Reply:
technically, identifiers starting with '_' and then an upper-case letter are reserved.

https://en.cppreference.com/w/cpp/language/identifiers